lua/telescope/previewers.lua

```diff
@@ -167,6 +176,41 @@ previewers.new = function(...)
   return Previewer:new(...)
 end
 
+previewers.new_buffer_previewer = function(opts)
+  opts = opts or {}
+
+  assert(opts.preview_fn, "preview_fn is required function")
+
+  if not opts.scroll_fn then
+    function opts.scroll_fn(self, direction)
+      local input = direction > 0 and "d" or "u"
+      local count = math.abs(direction)
+
+      self:send_input({ count = count, input = input })
+    end
+  end
+
+  if not opts.send_input then
+    function opts.send_input(self, input)
+      if not self.state then
+        return
+      end
+
+      local bufnr = vim.api.nvim_win_get_buf(self.state.hl_win)
+      local max_line = vim.fn.getbufinfo(bufnr)[1].linecount
+      local line = vim.api.nvim_win_get_cursor(self.state.hl_win)[1]
+      if input.input == 'u' then
+        line = (line - input.count) > 0 and (line - input.count) or 1
+      else
+        line = (line + input.count) <= max_line and (line + input.count) or max_line
+      end
+      vim.api.nvim_win_set_cursor(self.state.hl_win, { line, 1 })
```

Comment:
I'm still not happy about this "thing". I would love to just do `vim.api.nvim_feedkeys(key, 'n', true)`, but i am in the wrong window and `:exe nr .. wincmd w` does not work. Same goes for `windo`. Do you have any idea?

Reply:
Later bfreld and I will add `vim.api.nvim_ex_call` which will allow setting this like `noautocmd`, `windo` etc.

As of now, this is probably best

Reply:
I just looked at the pr. Looks really interesting. Will keep an eye on that one. Thanks for the review